eng/tools/rush-runner/index.js

```diff
@@ -69,160 +46,78 @@ const parseArgs = () => {
- * and not all of its dependents.
- * @param packageNames string[] An array of strings containing the packages names to run the action on.
+ * @param {string} cwd - current working directory
+ * @param {string[]} args - rest of arguments
  */
-const getDirectionMappedPackages = (packageNames) => {
-  const mappedPackages = [];
-
-  for (const packageName of packageNames) {
-    // Build command without any additional option should build the project and downstream
-    // If service is configured to run only a set of downstream projects then build all projects leading to them to support testing
-    // If the package is a core package, azure-identity or arm-resources then build only the package,
-    // otherwise build the package and all its dependents
-    var rushCommandFlag = "--impacted-by";
-
-    if (restrictedToPackages.includes(packageName)) {
-      // if this is one of our restricted packages with a ton of deps, make it targeted
-      // as including all dependents will be too much
-      rushCommandFlag = "--to";
-    } else if (actionComponents.length == 1) {
-      // else we are building the project and its dependents
-      rushCommandFlag = "--from";
-    }
-
-    mappedPackages.push([rushCommandFlag, packageName]);
-  }
-
-  return mappedPackages;
-};
-
-const getServicePackages = (baseDir, serviceDirs, artifactNames) => {
-  const packageNames = [];
-  const packageDirs = [];
-  let validSdkTypes = ["client", "mgmt", "perf-test", "utility"]; // valid "sdk-type"s that we are looking for, to be able to apply rush-runner jobs on
-  console.log(`Packages to build: ${artifactNames}`);
-  const artifacts = artifactNames.split(",");
-  for (const serviceDir of serviceDirs) {
-    const searchDir = path.resolve(path.join(baseDir, "sdk", serviceDir));
-    const packageJSONs = getPackageJSONs(searchDir);
-    for (const filePath of packageJSONs) {
-      const contents = JSON.parse(fs.readFileSync(filePath, "utf8"));
-      const artifactName = contents.name.replace("@", "").replace("/", "-");
-      if (
-        validSdkTypes.includes(contents["sdk-type"]) &&
-        (artifactNames.length === 0 || artifacts.includes(artifactName))
-      ) {
-        packageNames.push(contents.name);
-        packageDirs.push(path.dirname(filePath));
-      }
-    }
-  }
-  console.log(`Packages eligible to run rush task: ${packageNames}`);
-  return [packageNames, packageDirs];
-};
-
 const spawnNode = (cwd, ...args) => {
   console.log(`Executing: "node ${args.join(" ")}" in ${cwd}\n\n`);
   const proc = spawnSync("node", args, { cwd, stdio: "inherit" });
   console.log(`\n\nNode process exited with code ${proc.status} `);
 
-  if (proc.status !== 0) {
-    // proc.status will be null if the subprocess terminated due to a signal, which I don't think
-    // should ever happen, but if it does it's safer to fail.
-    process.exitCode = proc.status || 1;
```

Comment:
This was causing runtime error of 
![image](https://github.com/user-attachments/assets/fac48111-ed2c-4340-9221-d794970ec987)